twisted/test/test_policies.py

```diff
@@ -870,3 +876,43 @@ def test_counter(self):
         f.resetCounter()
         self.assertEqual(f._counter, 0)
 
+
+    def test_loggingFactoryOpensLogfileAutomatically(self):
+        """
+        When the L{policies.TrafficLoggingFactory} builds a protocol, it
+        automatically opens a unique log file for that protocol and attaches
+        the logfile to the built protocol.
+        """
+        open_calls = []
+        open_rvalues = []
+
+        def mocked_open(*args, **kwargs):
+            """
+            Mock for the open call to prevent actually opening a log file.
+            """
+            open_calls.append((args, kwargs))
+            io = NativeStringIO()
+            io.name = args[0]
+            open_rvalues.append(io)
+            return io
+
+        self.patch(builtins, 'open', mocked_open)
+
+        wrappedFactory = protocol.ServerFactory()
+        wrappedFactory.protocol = SimpleProtocol
+        factory = policies.TrafficLoggingFactory(wrappedFactory, 'test')
+        first_proto = factory.buildProtocol(address.IPv4Address('TCP',
+                                                                '127.0.0.1',
+                                                                12345))
```

Comment:
also minor comment. I find this indentation style bad suited as you end up squeezing all the data at the end of the line.

for me this looks better 
```
first_proto = factory.buildProtocol(
    address.IPv4Address('TCP', '127.0.0.1', 12345))
```